domains/client/domain-executor/src/core_bundle_processor.rs

```diff
@@ -205,9 +202,17 @@ where
     fn bundles_to_extrinsics(
         &self,
         parent_hash: Block::Hash,
-        bundles: CoreBundles<Block, PBlock>,
+        bundles: DomainBundles<Block, PBlock>,
         shuffling_seed: Randomness,
     ) -> Result<Vec<Block::Extrinsic>, sp_blockchain::Error> {
+        let bundles = match bundles {
+            DomainBundles::System(..) => {
+                return Err(sp_blockchain::Error::Application(Box::from(
+                    "Core bundle processor can not process system bundles.",
+                )))
+            }
+            DomainBundles::Core(bundles) => bundles,
+        };
```

Comment:
It is a bit awkward to have an enum, but only accepting one variant. Would be better to design an API (if possible) where you can't provide unsupported input at all. Otherwise it reduces amount of code, but increases complexity and maintainability.

Reply:
I agree it's not a good design and it's not impossible to have a better one, but that definitely requires much more refactoring work. Considering the increased maintainability is not significant, I'd love to take it for now and save time for the work with higher priority.